core/services/ocr2/plugins/ocr2keeper/evm21/streams_lookup.go

```diff
@@ -235,7 +235,7 @@ func (r *EvmRegistry) allowedToUseMercury(opts *bind.CallOpts, upkeepId *big.Int
 	}
 
 	// call checkCallback function at the block which OCR3 has agreed upon
-	err = r.client.CallContext(opts.Context, &resultBytes, "eth_call", args, opts.BlockNumber)
+	err = r.client.CallContext(opts.Context, &resultBytes, "eth_call", args, hexutil.EncodeUint64(opts.BlockNumber.Uint64()))
```

Comment:
can we use `hexutil.EncodeBig` as is used in other places

Reply:
I think both are being used in different place of the code. Uint makes more sense to me since negative numbers aren't valid blocks anyway.

Reply:
the other parts of code are within the same check pipeline flow https://github.com/smartcontractkit/chainlink/blob/develop/core/services/ocr2/plugins/ocr2keeper/evm21/registry_check_pipeline.go#L212-L219 so i would prefer consistency

Please thoroughly test this e2e before merging

Reply:
Sounds good.